packages/flutter/lib/src/material/slider.dart

```diff
@@ -907,152 +918,113 @@ class _RenderSlider extends RenderBox {
+    final List<double> trackDividers = <double>[visualPosition];
 
-    final double trackVerticalCenter = offset.dy + (size.height) / 2.0;
-    final double trackLeft = offset.dx + _overlayRadius;
-    final double trackTop = trackVerticalCenter - trackRadius;
-    final double trackBottom = trackVerticalCenter + trackRadius;
-    final double trackRight = trackLeft + trackLength;
-    final double trackActive = trackLeft + trackLength * visualPosition;
-    final double thumbRadius = _sliderTheme.thumbShape.getPreferredSize(isInteractive, isDiscrete).width / 2.0;
-    final double trackActiveLeft = math.max(0.0, trackActive - thumbRadius - thumbGap * (1.0 - _enableAnimation.value));
-    final double trackActiveRight = math.min(trackActive + thumbRadius + thumbGap * (1.0 - _enableAnimation.value), trackRight);
-    final Rect trackLeftRect = Rect.fromLTRB(trackLeft, trackTop, trackActiveLeft, trackBottom);
-    final Rect trackRightRect = Rect.fromLTRB(trackActiveRight, trackTop, trackRight, trackBottom);
+    // Paint the track.
+    _sliderTheme.trackShape.paint(context, offset,
+        parentBox: this,
+        sliderTheme: _sliderTheme,
+        enableAnimation: _enableAnimation,
+        trackDividers: trackDividers,
+        trackSegmentsActive: trackSegmentsActive,
+        isDiscrete: isDiscrete,
+        isEnabled: isInteractive
+    );
 
-    final Offset thumbCenter = Offset(trackActive, trackVerticalCenter);
+    final Rect trackRect = _sliderTheme.trackShape.getPreferredRect(
+      parentBox: this,
+      offset: offset,
+      sliderTheme: _sliderTheme,
+      isEnabled: isInteractive,
+      isDiscrete: isDiscrete
+    );
+    final Offset thumbCenter = Offset(trackRect.left + visualPosition * trackRect.width, trackRect.center.dy);
 
-    // Paint the track.
-    if (visualPosition > 0.0) {
-      canvas.drawRect(trackLeftRect, leftTrackPaint);
-    }
-    if (visualPosition < 1.0) {
-      canvas.drawRect(trackRightRect, rightTrackPaint);
+    // Paint the overlay.
+    if (!_overlayAnimation.isDismissed) {
+      _sliderTheme.overlayShape.paint(context, thumbCenter,
+        activationAnimation: _overlayAnimation,
+        enableAnimation: _enableAnimation,
+        isDiscrete: isDiscrete,
+        labelPainter: _labelPainter,
+        parentBox: this,
+        sliderTheme: _sliderTheme,
+        textDirection: _textDirection,
+        value: _value,
+      );
     }
 
-    _paintOverlay(canvas, thumbCenter);
-
-    _paintTickMarks(
-      canvas,
-      trackLeftRect,
-      trackRightRect,
-      leftTickMarkPaint,
-      rightTickMarkPaint,
-    );
+    // Paint the ticks.
+    if (isDiscrete) {
+      // TODO(clocksmith): Align tick centers to ends of track by not subtracting diameter from length.
+      final double tickDiameter = _sliderTheme.tickShape.getPreferredSize(
+        isEnabled: isInteractive,
+        sliderTheme: _sliderTheme).width;
```

Comment:
Nit again: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#if-you-have-a-newline-after-some-opening-punctuation-match-it-on-the-closing-punctuation (newline + trailing comma)
```
final double tickDiameter = _sliderTheme.tickShape.getPreferredSize(
  isEnabled: isInteractive,
  sliderTheme: _sliderTheme,
).width;
```